modules/core/src/main/java/org/apache/ignite/spi/tracing/SpanStatus.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.spi.tracing;
+
+/**
+ * Various statuses for Spans execution.
+ */
+public enum SpanStatus {
```

Comment:
Can you, please, clarify, why do we need this statuses?

Reply:
It's the concept, that we've borrowed from OpenCensus.

> Status defines a logical error model to represent a filterable state of the span at a point in time.
> Status allows tracing visualization tools to highlight unsuccessful spans and helps in debugging errors.

https://opencensus.io/tracing/span/status/
In Ignite statuses are used, for example, within discovery handling.
```
            if (locNodeId.equals(node.id())) {
                if (log.isDebugEnabled())
                    log.debug("Received join request for local node, dropping: " + msg);

                msg.spanContainer().span()
                    .addLog(() -> "Dropped")
                    .setStatus(SpanStatus.ABORTED)
                    .end();

                return;
            }
```
Besides other advantages statuses are used for automatic markup in trace visualization tools.